### PR TITLE
Prey System Update

### LINKED ISSUE #997: Willcard in prey not working.
if someone get willcards for tibia coins in store, when you want to change a creature just can do thats for money, willcard not working, can't choice another creatures with thats cards.

as

### LINKED ISSUE #1171: Re-rolling the bonus on the prey system
## CAUTION: OUR SERVER IS UPDATED DAILY. MAKE SURE YOU ARE REPORTING SOMETHING ON THE LATEST VERSION OF THE SERVER.

**Describe the bug**
When you re-roll the prey system bonus with wildcards the number of stars (bonus) moves back to less stars as you keep re-rolling. When it only should be getting better number of bonus as you keep rolling bonuses.

**Expected behavior**
When you re-roll prey bonus on the prey system the bonus should only be going up.
Ex.: If i have Loot Bonus 3 stars and i re-roll the bonus, either i can get Exp/Loot/Attack/Defence 4 stars or higher on the next re-roll. And so on, the re-roll always has to be better than the previous stars (like real tibia).

### LINKED ISSUE #1180: Prey System - All Problems till now.
1- We houldnt use a "rank list" to randomize the creatures;
"
CONST_MONSTER_TIER_BRONZE = 0
CONST_MONSTER_TIER_SILVER = 1
CONST_MONSTER_TIER_GOLD = 2
CONST_MONSTER_TIER_PLATINUM = 3"

we should use all the creatures list.

2- The list should be remade, for example:
rotworm queen/draptor, arent monster that shouldnt be in the list.

2- The timer should decay when the player have a battlesign, independent of the creature/ if you attack it or not, the timer should decay.

but these 2, are differents:
if u have exp prey, only when u kill something. (these 2 , follow yours stamina time).
if u have loot prey, only when u kill something. (these 2 , follow yours stamina time).

image

3- These three Features, arent working:

image

4- When you have for example 10 stars:

image

if u buy a Bonus reroll, it should continue with 10 stars.
but this is happening:

image

### LINKED ISSUE #1284: Prey Pick no work
Click here to choose your new prey creature from all available prey creatures. etc
for 5 prey cards. This does not work maybe there was an updated file i missed?

### PR DESCRIPTION
**Pull request by FakeShinoda**
- Automatic BonusReroll/Lock released
- Selection with 5 wildcard released
- Fixed bonuses with wildcard

* Added migrations
* Added some variables for configuration

Fix #997
Fix #1180

### CODE REVIEW

**Comment by oalbsilva** [comment]
missed migration DB on ticks column

File: data/modules/scripts/prey_system/prey.lua [hunk 1]
```diff
@@ -209,9 +199,69 @@ function Player.getMinutesUntilFreeReroll(self, slot)
...
+function Player.setAutomaticBonus(self, slot)
+	local monster = self:getPreyCurrentMonster(slot)
+
+	-- Automatic Bonus Reroll
+	if self:getPreyTick(slot) == 1 and self:getPreyBonusRerolls() >= 1 then
+		self:setPreyBonusType(slot, self:getDiffBonus(slot))
+		self:setRandomBonusValue(slot, true, true)
+		self:setPreyBonusRerolls(self:getPreyBonusRerolls() - 1)
+		self:sendTextMessage(MESSAGE_EVENT_ADVANCE, string.format("Your %s's prey bonus was automatically rolled.", monster:lower()))
+		self:setPreyTimeLeft(slot, 7200) -- 2 hours
```
Comment (oalbsilva) [inline data/modules/scripts/prey_system/prey.lua]: its better put the time on config variable

**Comment by oalbsilva** [comment]
Tested here. Its Working!

**Comment by Geekbuddy** [comment]
Tested! Everything works! 👍

**Comment by majestyotbr** [comment]
Tested, working fine. Just need to add migration for tick column to prey_slots table.

**Comment by Geekbuddy** [comment]
> Tested, working fine. Just need to add migration for tick column to prey_slots table.

ill did add tick column with int(11) in prey_slots but nothing happens with the prey time its stands on 01:59.

File: data/startup/others/functions.lua [hunk 2]
```diff
@@ -148,7 +148,7 @@ end
 
 -- Functions that cannot be used in reload command, so they have been moved here
 -- Prey slots consumption
-function preyTimeLeft(player, slot)
+local function preyTimeLeft(player, slot)
```
Comment (oalbsilva) [inline data/startup/others/functions.lua]: why local function here?

**Comment by majestyotbr** [comment]
Tested again, it's working but the PreyTimeLeft need a fix.

**Comment by majestyotbr** [comment]
Tested, it's working fine. Thanks for your contribution!

**Review by majestyotbr — APPROVED** [review]

**Pull request merged**